userspace/u2f_app/u2f_transport.c

```diff
@@ -407,13 +409,13 @@ static void u2fhid_cmd_init(U2FHID_FRAME *f_p) {
   /* DATA + 16 = Capabilities flags */
   /* TODO: Yes/no? */
   response.init.data[16] = CAPFLAG_WINK | CAPFLAG_LOCK;
-  printf("Response Frame -> cid:%08lx cmd:%02x ", response.cid,
-          response.init.cmd);
-  printf("bcnth:%02x bcntl:%02x ", response.init.bcnth, response.init.bcntl);
-  printf("data:%0X%0X%0X%0X%0X%0X%0X%0X\n\n", response.init.data[0],
-          response.init.data[1], response.init.data[2], response.init.data[3],
-          response.init.data[4], response.init.data[5], response.init.data[6],
-          response.init.data[7]);
+  //printf("Response Frame -> cid:%08lx cmd:%02x ", response.cid,
+  //        response.init.cmd);
+  //printf("bcnth:%02x bcntl:%02x ", response.init.bcnth, response.init.bcntl);
+  //printf("data:%0X%0X%0X%0X%0X%0X%0X%0X\n\n", response.init.data[0],
+  //        response.init.data[1], response.init.data[2], response.init.data[3],
+  //        response.init.data[4], response.init.data[5], response.init.data[6],
+  //        response.init.data[7]);
```

Comment:
I suspect these printf statements should be removed.

If we still want them, we should probably define a "debug_printf" or similar that is a no-op by default and can be enabled through a debug option (`make devprogram`?).

Reply:
I agree that commented out printf statements are bad.

The issue I'm wrestling with is that when something goes wrong, you often want information from many different layers at the same time, but within a particular code path. I.e., you want debugging information about the control channel (EP0) across different parts of the stack, or you want data path information (EP1) across different parts of the stack. You can just reintroduce debug output each time, but this is a pain. So in the best case we have a set of debug flags you can turn on/off. You can see this in the three different debug macros within usb/mod.rs.

It's worth noting this particular printf() is from the original C code. :)

Reply:
Yeah, this is a more general problem that we should solve at some point. A better debugging system would let us leave the debug messages in the code but make them no-ops (that compile away entirely) by default.

Reply:
Just cut it.